app/backend/src/updateBaseImagesInS3.ts

```
@@ -12,25 +12,29 @@ import { getKeysFromS3 } from './getKeysFromS3';
 import { updateCommitStatus } from './updateCommitStatus';
 
 export const updateBaseImagesInS3 = async ({
-  hash,
+  commitHash,
+  diffId,
   bucket,
   owner,
   repo
 }: UpdateBaseImagesInput) => {
-  const reasonToPreventUpdate = await findReasonToPreventBaseImageUpdate(
-    owner,
-    repo,
-    hash
-  );
+  const reasonToPreventUpdate =
+    commitHash &&
+    (await findReasonToPreventBaseImageUpdate(owner, repo, commitHash));
   if (reasonToPreventUpdate) {
     throw new TRPCError({
       code: 'FORBIDDEN',
       message: reasonToPreventUpdate
     });
   }
-  const s3Paths = await getKeysFromS3(hash, bucket);
-  await replaceImagesInS3(s3Paths, bucket);
-  await updateCommitStatus({ owner, repo, hash });
+  const hash = commitHash ?? diffId;
+  if (hash) {
+    const s3Paths = await getKeysFromS3(hash, bucket);
+    await replaceImagesInS3(s3Paths, bucket);
+  }
```

I'd almost rather throw a TRPC error here if !hash